client/components/login/login.test.js

```diff
@@ -392,4 +392,56 @@ describe("<Login /> interactions", () => {
     const setIsActiveMock = wrapper.instance().props.setIsActive.mock;
     expect(setIsActiveMock.calls.length).toBe(1);
   });
+  it("should show error toast when server error", () => {
+    axios.mockImplementationOnce(() => {
+      return Promise.reject({
+        status: 500,
+        statusText: "Internal server error",
+        response: {
+          data: {
+            detail: "Internal server error",
+          },
+        },
+      });
+    });
+    const event = {preventDefault: () => {}};
+    const errorMethod = jest.fn();
+    dependency.toast = {
+      error: errorMethod,
+    };
+    return wrapper
+      .instance()
+      .handleSubmit(event)
+      .then(() => {
+        expect(wrapper.instance().props.authenticate.mock.calls.length).toBe(0);
+        expect(lastConsoleOutuput).not.toBe(null);
+        expect(errorMethod).toHaveBeenCalled();
+        expect(errorMethod).toBeCalledWith("Internal server error");
+        lastConsoleOutuput = null;
```

Comment:
let's use `afterEach` to do this kind of operation, which is cleaner and more reliable (this line may not be executed if the test fails).
https://github.com/openwisp/openwisp-wifi-login-pages/blob/ceba2a78d9b733c7473e7962ea641351dd5741fa/client/components/registration/registration.test.js#L76-L78